### PR TITLE
Allow tools without datasource

### PR DESCRIPTION
**Pull request by marcusschiesser**
Simplify code generation

### CODE REVIEW

File: templates/components/engines/typescript/agent/chat.ts [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+      }),
+    );
+  }
 
-  const externalTools = await ToolFactory.createTools(config);
+  try {
+    // add tools from config file if it exists
+    const config = JSON.parse(
+      await fs.readFile(path.join("config", "tools.json"), "utf8"),
+    );
```
[the remark this answers is not in the document]
Comment (marcusschiesser) [reply templates/components/engines/typescript/agent/chat.ts:35]: to support that this file doesn't exist (no tools selected) - I simplified the generator that we either always use the ContextChatEngine or the AgentChatEngine

File: templates/components/engines/python/agent/__init__.py [hunk 2]
```diff
@@ -11,11 +11,12 @@ def get_chat_engine():
     top_k = os.getenv("TOP_K", "3")
     tools = []
 
-    # Add query tool
+    # Add query tool if index exists
```
Comment (leehuwuj) [inline templates/components/engines/python/agent/__init__.py:14]: Does this mode work if the user does not have an index and no tools config as well?

File: templates/components/engines/typescript/agent/chat.ts [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
+    tools = tools.concat(await ToolFactory.createTools(config));
```
[the remark this answers is not in the document]
Comment (marcusschiesser) [reply templates/components/engines/typescript/agent/chat.ts:36]: good idea. i just kept it as as is it's probably a bit better to read

File: templates/components/engines/python/agent/__init__.py (hunk 2 above)
Comment (marcusschiesser) [reply templates/components/engines/python/agent/__init__.py:14]: yes, tested this

File: helpers/index.ts [hunk 4]
```diff
@@ -121,6 +123,14 @@ export const installTemplate = async (
     await installTSTemplate(props);
   }
 
+  // write configuration files
+  await writeLoadersConfig(props.root, props.dataSources, props.useLlamaParse);
```
Comment (leehuwuj) [inline helpers/index.ts:127]: TS does not support loaders right now, right? I think we should only generate loaders config for python only for now

Comment (marcusschiesser) [reply helpers/index.ts]: @leehuwuj thanks good idea!

**Pull request merged**